### ISSUE TITLE
isAuthenticated() can be trivially spoofed

### ISSUE THREAD

**Issue by hackel** [sahat/satellizer#762]
Currently, isAuthenticated() returns true when any random string is stored in the local storage key instead of a properly formatted JWT, allowing users to "log in" to a site by simply setting this key.  Granted, most sensibly-designed applications will not really function in this scenario, since they're retrieving protected data from an API, but it is still not ideal.

Shouldn't isAuthenticated() FAIL in the "Non-JWT token that looks like JWT" and "All other tokens" cases?  I'm not seeing any legitimate use-case for storing non-JWT data in that same specific key.

**Comment by teotwaki** [comment]
There isn't really a way around this. Anyone will always be able to modify the way the client side behaves.

Even if satellizer were to validate that it is a JWT, and it is signed, what key would it verify against? A key that is embedded in the app? So modifiable/replaceable by the user with sufficient intent. Alternatively, one could monkey patch the validation function to always return true, etc.

I believe satellizer behaves correctly. If you agree, @hackel, would you mind closing this issue? Alternatively, please explain what method you have in mind for improvement.

**Comment by hackel** [comment]
Of course, I realise we can't rely on the client side for security.  I ran into this problem as a developer when the localstorage key was somehow getting corrupted, yet it still showed me as logged in.  It proved to be a pain to debug.  My thinking was that at least if isAuthenticated failed in this situation, it would clue a developer in to a problem.  `true` just seems like the wrong choice of default value when checking if an authentication token is valid, unless I am missing some potential use-case.
